lib/routes/token.js

```diff
@@ -464,45 +469,51 @@ function generateIdToken(options) {
     claims['fxa-aal'] = options.aal;
     claims.acr = 'AAL' + options.aal;
   }
+  if (options.generateAtHash) {
+    claims.at_hash = util.generateAtHash(options.access.token);
+  }
   return ID_TOKEN_KEY.sign(claims);
 }
 
-function generateTokens(options) {
+function generateTokens (options) {
   // we always are generating an access token here
   // but depending on options, we may also be generating a refresh_token
-  var promises = {
-    access: db.generateAccessToken(options)
-  };
-  if (options.offline) {
-    promises.refresh = db.generateRefreshToken(options);
-  }
-  if (options.idToken) {
-    promises.idToken = generateIdToken(options);
-  }
-  return P.props(promises).then(function(result) {
-    var access = result.access;
-    var refresh = result.refresh;
-    var idToken = result.idToken;
-
-    var json = {
-      access_token: access.token.toString('hex'),
-      token_type: access.type,
-      scope: access.scope.toString()
-    };
-    if (options.authAt) {
-      json.auth_at = options.authAt;
-    }
-    json.expires_in = options.ttl;
-    if (refresh) {
-      json.refresh_token = refresh.token.toString('hex');
-    }
-    if (idToken) {
-      json.id_token = idToken;
-    }
-    if (options.keysJwe) {
-      json.keys_jwe = options.keysJwe;
-    }
-    return json;
-  });
+  return db.generateAccessToken(options)
+    .then((access) => {
+      options.access = access;
```

Comment:
Had to tweak this function to generate the `access_token` first, then use it to generate the `at_hash` claim.